composer.lock

```diff
@@ -1,11 +1,509 @@
 {
```

Comment:
It's a library! Libraries shouldn't commit a `composer.lock`!

The only time they'll get used is when you're working on the project itself and if you're including the library, we need to be happy that any version of the requested dependencies will work

Reply:
Is this correct? Consumers of the library will have their own composer.json and will generate their own project's composer.lock, so the purpose of the lockfile still stands doesn't it? The purpose is that it provides a common set of dependencies for the next developer, knowing that the unit tests passed at the point the lockfile was committed.

Reply:
Yup! A library has to be able to have faith that it works with the latest version of the dependencies it says it's willing to work with (`~2` etc).

If you're running it through CI, you *want* it to pull in the latest dependencies